sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/ServerBatchRequest.java

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation.batch;
+
+import com.azure.cosmos.implementation.JsonSerializable;
+import com.azure.cosmos.implementation.Utils;
+import com.azure.cosmos.implementation.apachecommons.collections.list.UnmodifiableList;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+
+import java.util.List;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkState;
+
+/**
+ * This class represents a server batch request.
+ */
+public abstract class ServerBatchRequest {
+
+    private final int maxBodyLength;
+    private final int maxOperationCount;
+
+    private String requestBody;
+    private List<ItemBatchOperation<?>> operations;
+    private boolean isAtomicBatch = false;
+    private boolean shouldContinueOnError = false;
+
+    /**
+     * Initializes a new {@link ServerBatchRequest request} instance.
+     *
+     * @param maxBodyLength Maximum length allowed for the request body.
+     * @param maxOperationCount Maximum number of operations allowed in the request.
+     */
+    ServerBatchRequest(int maxBodyLength, int maxOperationCount) {
+        this.maxBodyLength = maxBodyLength;
+        this.maxOperationCount = maxOperationCount;
+    }
+
+    /**
+     * Adds as many operations as possible from the given list of operations.
+     * TODO(rakkuma): Similarly for hybrid row, request needs to be parsed to create a request body in any form.
+     * Issue: https://github.com/Azure/azure-sdk-for-java/issues/15856
+     *
+     * Operations are added in order while ensuring the request body never exceeds {@link #maxBodyLength}.
+     *
+     * @param operations operations to be added; read-only.
+     *
+     * @return Any pending operations that were not included in the request.
+     */
+    final List<ItemBatchOperation<?>> createBodyOfBatchRequest(final List<ItemBatchOperation<?>> operations) {
+
+        checkNotNull(operations, "expected non-null operations");
+
+        int totalSerializedLength = 0;
+        int totalOperationCount = 0;
+
+        final ArrayNode arrayNode =  Utils.getSimpleObjectMapper().createArrayNode();
+
+        for(ItemBatchOperation<?> operation : operations) {
+
+            final JsonSerializable operationJsonSerializable = ItemBatchOperation.writeOperation(operation);
+            final int operationSerializedLength = operationJsonSerializable.toString().length();
```

Comment:
FYI, this is not the length byte, if the string contains unicode the byte encoding len will be more than this.

Reply:
+1  - i think it is ok fixing this later - outside the scope of this PR - but please track it via a GitHub item and link the ID here

Reply:
Yeah. Created a issue and added the link here. Thanks.